### DISCUSSION TITLE
Feature Request: case insensitive for layer name, st_read() Excel files

### DISCUSSION THREAD

**Discussion by frankjar** [duckdb/duckdb-spatial#350] [Ideas]
Why do you want this feature?

Excel can't have two sheets with the same case insensitive name : 'Sheet1' and 'sheet1'.

ST_READ() function of the spatial extension can specify a sheet with the layer parameter but naming the sheet is case insensitive :

```
select * from st_read('test.xlsx',layer='sheet1');
Binder Error: Layer 'sheet1' could not be found in dataset
```
And using 'Sheet1' instead of 'sheet1' works fine :
```
select * from st_read('test.xlsx',layer='Sheet1');
┌─────────┬─────────┐
│  COL1   │  COL2   │
│ varchar │ varchar │
├─────────┼─────────┤
│ A       │ A       │
│ B       │ B       │
└─────────┴─────────┘
```

Thank you for your interest for the feature.

**Comment by Maxxen** [comment]
Hello!

As of DuckDB 1.2.0 the duckdb [excel extension](https://duckdb.org/docs/extensions/excel.html#reading-xlsx-files) provides support for reading and writing xlsx files with greater efficiency and many more options. Therefore we're going to deprecate this feature in the spatial extension in the future. I would recommend that you try out the excel extension, and if you run into any problems please open an issue over at the [duckdb-excel repository](https://github.com/duckdb/duckdb-excel).
